Modules/_sqlite/connection.c

```diff
@@ -2173,6 +2173,45 @@ pysqlite_connection_create_collation_impl(pysqlite_Connection *self,
     Py_RETURN_NONE;
 }
 
+/*[clinic input]
+_sqlite3.Connection.set_file_control as pysqlite_connection_set_file_control
+
+    op: int
+      a SQLITE_FCNTL_ constant
+    arg: long
+      argument to pass
+    /
+    dbname: str = NULL
+      database name
+
+Invoke a file control method on the database.
+[clinic start generated code]*/
+
+static PyObject *
+pysqlite_connection_set_file_control_impl(pysqlite_Connection *self, int op,
+                                          long arg, const char *dbname)
+/*[clinic end generated code: output=d9d2d311892893b6 input=0253798d9514fea2]*/
+{
+    int rc;
+    long val = arg;
+
+    if (!pysqlite_check_thread(self) || !pysqlite_check_connection(self)) {
+        return NULL;
+    }
+
+    Py_BEGIN_ALLOW_THREADS
+    rc = sqlite3_file_control(self->db, dbname, op, &val);
+    Py_END_ALLOW_THREADS
+
+    if (rc != SQLITE_OK) {
+        PyErr_SetString(self->ProgrammingError, sqlite3_errstr(rc));
```

Comment:
`ProgrammingError` may not be the right fit for all opcodes. Ideally, we should use a variant of `_pysqlite_seterror` where we pass an error code instead of a SQLite database pointer. Perhaps I should make such an internal API available first.

Reply:
I'd be happy to use any internal API that you build for this purpose.

Reply:
Give me a couple of days; my CPython bandwidth is pretty low this week, but I'll try to cook something up within the week. I'll try to jot down my plan in a dedicated issue soon.

Reply:
Draft PR up: #129929